Resources/Catalog/FileCatalog.py

```diff
@@ -93,57 +144,86 @@ def w_execute( self, *parms, **kws ):
     """
     successful = {}
     failed = {}
-    failedCatalogs = []
-    fileInfo = parms[0]
-    res = checkArgumentFormat( fileInfo )
-    if not res['OK']:
-      return res
-    fileInfo = res['Value']
-    allLfns = fileInfo.keys()
-    parms1 = parms[1:]
-    lfnsFlag = False
+    failedCatalogs = {}
+    successfulCatalogs = {}
+
+    allLfns = []
+    lfnMapDict = {}
+    masterResult = {}
+    if not self.call in FileCatalog.no_lfn_methods:
+      fileInfo = parms[0]
+      result = checkArgumentFormat( fileInfo, generateMap = True )
+      if not result['OK']:
+        return result
+      fileInfo, lfnMapDict = result['Value']
+      # No need to check the LFNs again in the clients
+      kws['LFNChecking'] = False
+      allLfns = fileInfo.keys()
+      parms1 = parms[1:]
+
     for catalogName, oCatalog, master in self.writeCatalogs:
 
-      # Skip if metadata related method on pure File Catalog
-      if self.call in FileCatalog.write_meta_methods and not catalogName in self.metaCatalogs:
-        continue
+      # Skip if the method is not implemented in this catalog
+      if not oCatalog.hasCatalogMethod( self.call ):
+        if master:
+          gLogger.error( "Master catalog does not implement the write method", self.call )
+          return S_ERROR( "Master catalog does not implement the write method %s" % self.call )
```

Comment:
OK, sorry... I had not realized it was for master only... 
B.t.w. there should be somewhere a definition of what a "master" is, i.e. specific actions taken when dealing with a master, but this is for the documentation...